.github/workflows/build-install-test-snap.yaml

```diff
@@ -87,3 +66,51 @@ jobs:
       with:
         name: ${{ inputs.snap-name }}-${{ steps.branch-name.outputs.BRANCH_NAME }}
         path: ${{ steps.build-snap.outputs.snap }}
+  
+  test:
+    runs-on: '${{ inputs.ubuntu-image }}'
+    needs: build
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.snap-name }}-${{needs.build.outputs.upload-branch-name}}
+          path: ${{ inputs.snapcraft-build-path }}
+      - name: Install snap with flags
+        if: inputs.snap-install-args != ''
+        run: |
+          sudo snap install ${{needs.build.outputs.snap-file}} '${{ inputs.snap-install-args }}'
+      - name: Install snap without flags
+        if: inputs.snap-install-args == ''
+        run: |
+          sudo snap install ${{needs.build.outputs.snap-file}}
+      - name: Run custom test script
+        if: inputs.test-script != ''
+        shell: bash
+        run: |
+          cat > test-script.sh << 'EOF'
+            ${{ inputs.test-script }}
+          EOF
+          bash test-script.sh
+      - name: Test info snap
+        if: inputs.test-script == ''
+        run: |
+          snap info ${{ inputs.snap-name }}
+
+  publish:
+    if: inputs.publish == true
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.snap-name }}-${{needs.build.outputs.upload-branch-name}}
+        path: .
+  
+    - uses: snapcore/action-publish@v1
+      env:
+        SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+      if: env.SNAPCRAFT_STORE_CREDENTIALS
+      with:
+        snap: ${{needs.build.outputs.snap-file}}
+        release: ${{ startsWith(github.ref, 'refs/tags/') && 'candidate' || 'edge'}}
```

Comment:
I'd suggest to add a last job to delete the artifact. This job could be optional (default to true),

```yaml
cleanup:
    runs-on: ubuntu-latest
    needs: [publish, test]
    steps:
      - uses: geekyeggo/delete-artifact@v4
        with:
            name: snap
```